test/preprocessing/geometries/geometries.jl

```diff
@@ -0,0 +1,186 @@
+
+                geometry = load_geometry(joinpath(data_dir, files[i] * ".asc"))
+
+                @test TrixiParticles.nfaces(geometry) == n_edges[i]
+
+                @testset "Normals $j" for j in eachindex(geometry.vertex_normals)
+                    @test isapprox(geometry.vertex_normals[j][1], vertex_normals[:, j],
+                                   atol=1e-4)
+                end
+                @testset "Points $j" for j in eachindex(geometry.vertices)[1:(end - 1)]
+                    @test isapprox(geometry.vertices[j], points[:, j], atol=1e-4)
+                end
+            end
+        end
+        @testset verbose=true "3D" begin
+            files = ["sphere", "bar", "gear"]
+            n_faces = [192, 12, 12406]
+            n_vertices = [98, 8, 6203]
+
+            @testset "Test File `$(files[i])`" for i in eachindex(files)
+                # Checked in ParaView with `trixi2vtk(geometry)`
+                data = TrixiParticles.CSV.read(joinpath(validation_dir(), "preprocessing",
+                                                        "normals_" * files[i] * ".csv"),
+                                               TrixiParticles.DataFrame)
+
+                vertex_normals = reinterpret(reshape, SVector{3, Float64},
+                                             vcat((data.var"vertex_normals:0")',
+                                                  (data.var"vertex_normals:1")',
+                                                  (data.var"vertex_normals:2")'))
+
+                points = reinterpret(reshape, SVector{3, Float64},
+                                     vcat((data.var"Points:0")',
+                                          (data.var"Points:1")',
+                                          (data.var"Points:2")'))
+
+                geometry = load_geometry(joinpath(data_dir, files[i] * ".stl"))
+
+                @test TrixiParticles.nfaces(geometry) == n_faces[i]
+                @test length(geometry.vertices) == n_vertices[i]
+
+                @testset "Normals $j" for j in eachindex(geometry.vertices)
+                    @test isapprox(geometry.vertex_normals[j], vertex_normals[j], atol=1e-5)
+                end
+
+                @testset "Points $j" for j in eachindex(geometry.vertices)
+                    @test isapprox(geometry.vertices[j], points[j], atol=1e-4)
+                end
+            end
+        end
+    end
+
+    @testset verbose=true "Unique Sort" begin
+        # Fixed seed to ensure reproducibility
+        Random.seed!(1)
+
+        x = rand(SVector{3, Float64}, 10_000)
+
+        # Make sure that `x` is not unique
+        for i in eachindex(x)[2:end]
+            # Create duplicate values
+            if rand(Bool)
+                x[i] = x[i - 1]
+            end
+        end
+
+        @test TrixiParticles.unique_sorted(copy(x)) == sort(unique(x))
+    end
+end
+
+# ==========================================================================================
```

Comment:
I don't think it's a good idea to leave code commented out that might be used later. Once we start doing this, it usually escalates and results in a lot of unused code. How about leaving it in a stash or in a draft PR until you need it?